remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java

```diff
@@ -2763,6 +2766,38 @@ public Node updateContent(String fileNodeId, BasicContentInfo contentInfo, Input
         return updateExistingFile(null, nodeRef, fileName, contentInfo, stream, parameters, versionMajor, versionComment);
     }
 
+    private void setSiteManagerPermission(NodeRef nodeRef, NodePermissions nodePerms)
+    {
+        if (nodeRef != null && nodePerms != null)
+        {
+            try
+            {
+                if (nodePerms.getIsInheritanceEnabled() != null && !nodePerms.getIsInheritanceEnabled())
+                {
+                    SiteInfo containingSite = siteService.getSite(nodeRef);
+
+                    if (containingSite != null)
+                    {
+                        String thisSiteGroupPrefix = siteService.getSiteGroup(containingSite.getShortName());
+                        final String siteManagerAuthority = thisSiteGroupPrefix + "_" + SiteModel.SITE_MANAGER;
+                        AuthenticationUtil.runAs(new AuthenticationUtil.RunAsWork<Void>()
+                        {
+                            public Void doWork() throws Exception
```

Comment:
The `runAs` can be replaced by a lambda function if you want.

Reply:
Replaced the `runAs` call as suggested.